plat/kvm/irq.c

```diff
@@ -86,13 +89,24 @@ int ukplat_irq_register(unsigned long irq, irq_handler_func_t func, void *arg)
  */
 extern unsigned long sched_have_pending_events;
 
-void _ukplat_irq_handle(unsigned long irq)
+void _ukplat_irq_handle(struct __regs *regs, unsigned long irq)
 {
 	struct irq_handler *h;
 	int i;
+	int rc;
+	struct ukplat_event_irq_data ctx;
 
 	UK_ASSERT(irq < __MAX_IRQ);
 
+	ctx.regs = regs;
+	ctx.irq = irq;
+	rc = uk_raise_event(UKARCH_EVENT_IRQ, &ctx);
+	if (rc == UK_EVENT_HANDLED)
```

Comment:
Since you're not using `rc` anywhere else, why not remove it and just
```
if (uk_raise_event(UKARCH_EVENT_IRQ, &ctx) == UK_EVENT_HANDLED)
```

Reply:
Personally, I prefer the separate `rc` version just for consistency with the rest of the code, but   add a `likely` for good practice. And with the `likely` also the line becomes a bit long and hard to read.

Reply:
I don't see a reason to use either `likely` or `unlikely` here. The return values really depends on the event handlers and the parameters. (But there is still the error case `rc < 0` missing, which is introduced in the other PR, for that unlikely makes sense).

For the comments on the `rc` variable I agree with Marc.